tests/utils.go

```diff
@@ -320,6 +339,136 @@ func BeforeTestSuitSetup() {
+	if err != nil {
+		panic(err)
+	}
+	matchFileName := func(pattern, filename string) bool {
+		match, err := filepath.Match(pattern, filename)
+		if err != nil {
+			panic(err)
+		}
+		return match
+	}
+	for _, file := range files {
+		if matchFileName("*-for-ocp.yaml", file.Name()) {
+			if CurrentClusterType.value == OpenShiftClusterType {
+				manifests = append(manifests, filepath.Join(pathToManifestsDir, file.Name()))
+			}
+		} else if matchFileName("*-for-k8s.yaml", file.Name()) {
+			if CurrentClusterType.value == KubernetesClusterType {
+				manifests = append(manifests, filepath.Join(pathToManifestsDir, file.Name()))
+			}
+		} else if matchFileName("*.yaml", file.Name()) {
+			manifests = append(manifests, filepath.Join(pathToManifestsDir, file.Name()))
+		}
+	}
+	return manifests
+}
+
+func ReadManifestYamlFile(pathToManifest string) []map[string]string {
+	// Read manifest
+	content, err := ioutil.ReadFile(pathToManifest)
+	if err != nil {
+		panic(err)
+	}
+	// Split manifest by `---` to get single objects
+	fileAsString := string(content[:])
+	sepYamlfiles := strings.Split(fileAsString, "---")
+	// Parse each object
+	objects := make([]map[string]string, 0, len(sepYamlfiles))
+	for _, f := range sepYamlfiles {
+		if f == "\n" || f == "" {
+			// ignore empty cases
+			continue
+		}
+
+		var obj map[interface{}]interface{}
+		err := yaml.Unmarshal([]byte(f), &obj)
+		if err != nil {
+			fmt.Printf(fmt.Sprintf("ERROR: %s\n", err))
+		}
+		fmt.Printf(fmt.Sprintf("  Kind(%s)\n", obj["kind"]))
+		fmt.Printf(fmt.Sprintf("    %s\n", obj))
+
+		item := make(map[string]string)
+		item["kind"] = obj["kind"].(string)
+		item["apiVersion"] = obj["apiVersion"].(string)
+		metadata := obj["metadata"].(map[interface{}]interface{})
+		if metadata != nil {
+			if metadata["namespace"] != nil {
+				item["namespace"] = metadata["namespace"].(string)
+			}
+		} else {
+			item["namespace"] = ""
+		}
+		item["data"] = f
+		objects = append(objects, item)
+	}
+
+	return objects
+}
+
+func GetResourceAPIEntryPoint(kind string) string {
+	var apiPoint string
+	switch kind {
+	case "ClusterRoleBinding":
+		apiPoint = "clusterrolebindings"
+	case "ClusterRole":
+		apiPoint = "clusterrolebindings"
+	case "ServiceAccount":
+		apiPoint = "serviceaccounts"
+	case "PersistentVolumeClaim":
+		apiPoint = "persistentvolumeclaims"
+	case "PersistentVolume":
+		apiPoint = "persistentvolumes"
+	case "DaemonSet":
+		apiPoint = "daemonsets"
+	case "SecurityContextConstraints":
+		apiPoint = "securitycontextconstraints"
+	default:
+		panic(fmt.Errorf("Can not map kind %s to api entry point - unknown kind", kind))
+	}
+	return apiPoint
+}
+
+func ApplyRawManifest(apiVersion, namespace, kind string, manifestData []byte) error {
+	virtCli, err := kubecli.GetKubevirtClient()
+	PanicOnError(err)
+	var request *k8srest.Request
+	if apiVersion == "v1" {
+		request = virtCli.CoreV1().RESTClient().Post()
+	} else {
+		request = virtCli.RestClient().Post().Prefix(apiVersion)
```

Comment:
I was able to get the apiVersion from manifest but I am not sure how to set APIGroup properly for DaemonSet or `ClusterRoleBinding`.